docs/docs/forms.mdx

```diff
@@ -360,74 +360,104 @@ to fully customize the validation process.
 
 If none of the predefined [Slot Mappings](./forms.mdx#slot-mappings) fit your use
 case, you can use the
-[Custom Action](./actions.mdx#custom-actions) `validate_{form_name}` to write your own
-extraction code. Rasa Open Source will trigger this function when the form is run.
-
-Make sure your custom action returns `SlotSet` events for every extracted value.
-The following example shows the implementation of a custom slot mapping which sets
-a slot based on the length of the last user message.
+[Custom Action](./actions.mdx#custom-actions) `validate_<form_name>` to write your own
+extraction code. Rasa Open Source will trigger this action when the form is run.
+
+If you're using the Rasa SDK we recommend you to extend the provided
+`FormValidationAction`. When using the `FormValidationAction`, three steps are required
+to extract customs slots:
+
+1. Define a method `extract_<slot_name>` for every slot that should be mapped in a
+   custom way.
+2. Make sure that in the domain file you list for your form only those slots that don't
+   use custom mappings.
```

Comment:
I'm thinking about using "predefined mapping" more often. @wochinge what do you think?
```suggestion
2. Make sure that in the domain file you list for your form only those slots that use 
[predefined mappings](./forms.mdx#slot-mappings).
```

Reply:
nice, I like it!